capture_service/device_mgr.cc

```diff
@@ -140,15 +140,24 @@ absl::StatusOr<std::vector<std::string>> AndroidDevice::ListPackage(PackageListO
     return package_list;
 }
 
-std::filesystem::path ResolveAndroidLibPath(const std::string &name)
+std::filesystem::path ResolveAndroidLibPath(const std::string &name,
+                                            const std::string &device_architecture)
 {
     LOGD("cwd: %s\n", std::filesystem::current_path().c_str());
     std::vector<std::filesystem::path> search_paths{ std::filesystem::path{ "./install" },
                                                      std::filesystem::path{
                                                      "../../build_android/Release/bin" },
                                                      std::filesystem::path{ "../../install" },
                                                      std::filesystem::path{ "./" } };
-    std::filesystem::path              lib_path{ name };
+
+    if (device_architecture != "")
```

Comment:
i guess device_architecture should never be ""? the capture would fail if it is ""?

Reply:
If it is an empty string then the gfxr layer will not be found. The device architecture is empty when running anything that is not gfxr capture.

Reply:
in our case, should we add some warning if this field is missing? (i am a bit worried about silent failures. )

Reply:
oh, i just see the default will be x86